packages/caliper-fisco-bcos/lib/fiscoBcos.js

```diff
@@ -0,0 +1,143 @@
+     * @param {Integer} clientIdx the client index
+     * @return {Promise<object>} The promise for the result of the execution.
+     */
+    async getContext(name, args, clientIdx) {
+        return Promise.resolve();
+    }
+
+    /**
+     * Release a context as well as related resources
+     * @param {Object} context adapter specific object
+     * @return {Promise<object>} The promise for the result of the execution.
+     */
+    async releaseContext(context) {
+        return Promise.resolve();
+    }
+
+    /**
+     * Invoke the given smart contract according to the specified options. Multiple transactions will be generated according to the length of args.
+     * @param {object} context The FISCO BCOS context returned by {getContext}.
+     * @param {string} contractID The name of the smart contract.
+     * @param {string} contractVer The version of the smart contract.
+     * @param {Array} args Array of JSON formatted arguments for transaction(s). Each element contains arguments (including the function name) passing to the smart contract. JSON attribute named transaction_type is used by default to specify the function name. If the attribute does not exist, the first attribute will be used as the function name.
+     * @param {number} timeout The timeout to set for the execution in seconds.
+     * @return {Promise<object>} The promise for the result of the execution.
+     */
+    async invokeSmartContract(context, contractID, contractVer, args, timeout) {
+        const fiscoBcosSettings = CaliperUtils.parseYaml(this.configPath)['fisco-bcos'];
+        let promises = [];
+        try {
+            args.forEach((arg) => {
+                let fcn = null;
+                let fcArgs = [];
+
+                for (let key in arg) {
+                    if (key === 'transaction_type') {
+                        fcn = arg[key].toString();
+                    } else {
+                        fcArgs.push(arg[key].toString());
+                    }
+                }
+                promises.push(invokeSmartContractImpl.run(context, fiscoBcosSettings, contractID, fcn, fcArgs, this.workspaceRoot));
+            });
+
+            return await Promise.all(promises);
+        } catch (error) {
+            commLogger.error(Color.error(`FISCO BCOS smart contract invoke failed: ${(error.stack ? error.stack : error)}`));
+            throw error;
+        }
+    }
+
+    /**
+     * Query state from the ledger
+     * @param {Object} context context object from getContext
+     * @param {String} contractID identity of the contract
+     * @param {String} contractVer version of the contract
+     * @param {String} key lookup key
+     * @param {String} fcn The smart contract query function name
+     * @return {any} The result of the query.
+     */
+    async queryState(context, contractID, contractVer, key, fcn) {
+        const fiscoBcosSettings = CaliperUtils.parseYaml(this.configPath)['fisco-bcos'];
```

Comment:
same comment as that in `invokeSmartContract`

Reply:
Solved :)